packages/common/src/content/isHostedAgolService.ts

```diff
@@ -0,0 +1,24 @@
+export function getHostname(url: string): string {
+  const urlObj = new URL(url);
+  return urlObj.hostname;
+}
+
+/* istanbul ignore next - fn copied from another location that has tests */
+export function isHostedAgolService(url: string): boolean {
+  let origin = getHostname(url); // -> www.esri.com
+
+  if (!origin) {
+    return false;
+  }
+
+  origin = origin.toLowerCase();
+
+  // unfortunately, when a service is proxied and requires credentials, we have no way of knowing if it's hosted or not
+  // ... meaning that we are returning false for all services that are proxied
```

Comment:
I can guarantee you that future devs will be confused by this comment since it assumes a lot of knowledge. Maybe we can expand this comment to be something like:
```
This function will return _true_ if the url points to a hosted service in ArcGIS Online. It will return _false_ for proxy services (i.e., a proxy set up by AGO to access secure services that require credentials). 

NOTE: Even if a proxy is delegating to hosted feature service, this util will return false. We have no way of identifying what kind of service a proxy represents.
```